### DISCUSSION TITLE
new functions - Correlation function and normalized cross correlation function required

### DISCUSSION THREAD

**Discussion by hamza-712** [v923z/micropython-ulab#597] [Ideas]
Hi, I have implemented my own correlation function but it leads to memory errors and its slow.
I wonder if you can include numpy.corrcoef() function

np.correlate(a, b, 'full')

Example normalized cross correlation
``` python
import numpy as np
a = [1, 2, 3, 4]
b = [2, 4, 6, 8]
norm_a = np.linalg.norm(a)
a = a / norm_a
norm_b = np.linalg.norm(b)
b = b / norm_b
c = np.correlate(a, b, mode = 'full')
```

**Comment by v923z** [comment]
@hamza-712 Yes, we could do that. Give me a week to pull this off.

**Reply by hamza-712** [reply]
Thanks alot
Can you include numpy.cov  too
Link is below
https://numpy.org/doc/stable/reference/generated/numpy.cov.html#numpy-cov

**Reply by v923z** [reply]
Have you looked at the comment in https://github.com/v923z/micropython-ulab/issues/600?

**Reply by hamza-712** [reply]
Yeah, I've looked at. Thanks for the insight.
